src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java

```diff
@@ -72,12 +93,30 @@ public void finishTree(DetailAST ast) {
             if (ast != null) {
                 lineNumber = ast.getLineNo();
             }
-            log(lineNumber, MSG_KEY);
+            log(lineNumber, MSG_KEY_MISSING);
         }
     }
 
     @Override
     public void visitToken(DetailAST ast) {
         defined = true;
+
+        if (!ignoreMismatch) {
+            // Calculate the directory name, but stripping off the last
```

Comment:
Try to avoid comments. Nobody will read them, nobody will trust them. Method with a clear name is much more better than comment. As you are trying to get directory name (lines 107 - 109) why don't you introduce a new method?